dotCMS/src/main/java/com/dotcms/storage/DataBaseStorage.java

```diff
@@ -0,0 +1,554 @@
+                    objectHash, path, groupName, metaDataJsonWriter.toString());
+
+            int order = 1;
+            for (final String chuckHash : chuckHashes) {
+
+                new DotConnect().executeUpdate(this.getConnection(),"insert into storage_x_data(storage_hash, data_hash, data_order) values (?, ?, ?)",
+                        objectHash, chuckHash, order++);
+            }
+
+            return true;
+        } catch (DotDataException | NoSuchAlgorithmException | IOException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotRuntimeException(e);
+        }
+    }
+
+    private  byte[]  chuckBytes(final int bytesLength, final byte[] bytes) {
+
+        final byte[] chunkedArray = new byte[bytesLength];
+
+        System.arraycopy(bytes, 0, chunkedArray, 0, bytesLength);
+
+        return chunkedArray;
+    }
+
+    private byte[] fileToBytes(final File file) {
+
+        try (InputStream input = com.liferay.util.FileUtil.createInputStream(file.toPath(), "none")) {
+
+            final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();  // todo not sure if I should compress this
+            final byte[] buffer = new byte[1024];
+            for (int readNum; (readNum = input.read(buffer)) != -1;) {
+
+                byteArrayOutputStream.write(buffer, 0, readNum);
+            }
+
+            return  byteArrayOutputStream.toByteArray();
+        } catch (IOException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotRuntimeException(e);
+        }
+    }
+
+    @Override
+    public Object pushObject(final String groupName, final String path,
+                             final ObjectWriterDelegate writerDelegate,
+                             final Serializable object, final Map<String, Object> extraMeta) {
+
+        try {
+
+            final File file                     = FileUtil.createTemporalFile("object-storage", ".tmp");
+            final byte[] objectBytes            = this.objectToBytes (writerDelegate, object);
+            FileUtils.writeByteArrayToFile(file, objectBytes);
+
+            return this.pushFile(groupName, path, file, extraMeta);
+        } catch (IOException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotRuntimeException(e);
+        }
+    }
+
+    private byte[] objectToBytes(final ObjectWriterDelegate writerDelegate,
+                                 final Serializable object) {
+
+        final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        writerDelegate.write(byteArrayOutputStream, object);
+        return  byteArrayOutputStream.toByteArray();
+    }
+
+    @Override
+    public Future<Object> pushFileAsync(final String bucketName, final String path, final File file, final Map<String, Object> extraMeta) {
+
+        return DotConcurrentFactory.getInstance().getSubmitter("StoragePool").submit(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "StoragePool" appears 4 times in this file; the first occurrence is on line 433](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)